app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -1,22 +1,66 @@
 package com.example.drawingApp
 
-import android.graphics.Bitmap
 import android.graphics.Color
-import androidx.annotation.ColorInt
 import androidx.lifecycle.ViewModel
 
 class DrawingViewModel : ViewModel() {
 
+    data class ViewState(
+        val isColorSheetOpen: Boolean,
+        val chosenColor: Int
+    )
 
-    @ColorInt
-    var primaryColor = Color.BLACK
+    private var isColorPickerSheetOpen = false
 
-    val layerNames: MutableList<String> = mutableListOf()
+    private var chosenColor: Int? = null
 
-    var activeBitmap: Bitmap? = null
+    private var hsv = Hsv(0f, 0f, 0f)
 
-    var isLayerCreationDialogOpen = false
-    var currentLayerCreationDialogText = ""
+    //used to help determine rgb values for consistency
+    var activeColor: Int = Color.BLACK
 
+    var onUpdate: ((ViewState) -> Unit)? = null
 
+    fun openColorSheet() {
+        isColorPickerSheetOpen = true
+        updateViewState()
+    }
+
+    fun closeColorSheet(isNotSubmitted: Boolean) {
+        isColorPickerSheetOpen = false
+        if (!isNotSubmitted) {
+            hsv = Hsv(0f, 0f, 0f)
+        }
+        updateViewState()
+    }
+
+    fun updateViewState() {
+        onUpdate?.invoke(
+            ViewState(
+                isColorSheetOpen = isColorPickerSheetOpen,
+                chosenColor = chosenColor ?: Color.BLACK
+            )
+        )
+    }
+
+    fun updateColorFromHsv() {
+        val color = hsv.toColor()
+        activeColor = color!!
```

Comment:
1. Why make a color variable just to set it a line later?
2. No need to double bang this

`activeColor = hsv.toColor() ?: activeColor` ?

Reply:
good point, remade it without a double bang.
I think the reason why it was like that previously was that the function did more previously and with pruning stuff was removed from it.